openpype/hosts/maya/plugins/publish/collect_render.py

```diff
@@ -221,22 +221,28 @@ def process(self, context):
             frame_end_render = int(self.get_render_attribute(
                 "endFrame", layer=layer_name))
 
+            handle_start = None
+            handle_end = None
+            if context.data.get("handles"):
+                handle_start = context.data.get("handles")
+                handle_end = context.data.get("handles")
+
             if (int(context.data['frameStartHandle']) == frame_start_render
                     and int(context.data['frameEndHandle']) == frame_end_render):  # noqa: W503, E501
 
-                handle_start = context.data['handleStart']
-                handle_end = context.data['handleEnd']
+                handle_start = context.data.get('handleStart') or handle_start
+                handle_end = context.data.get('handleEnd') or handle_end
                 frame_start = context.data['frameStart']
                 frame_end = context.data['frameEnd']
-                frame_start_handle = context.data['frameStartHandle']
-                frame_end_handle = context.data['frameEndHandle']
+
```

Comment:
Continuing on my comment. Basically this "if" branch - should not exist. And collect render should receive a way to potentially apply a custom handle start and end if needed.